lib/AppInfo/Application.php

```diff
@@ -53,6 +59,10 @@ public function register(IRegistrationContext $context): void {
 		$context->registerEventListener(LoadViewer::class, LoadViewerListener::class);
 		$context->registerEventListener(LoadAdditionalScriptsEvent::class, FilesLoadAdditionalScriptsListener::class);
 		$context->registerEventListener(BeforeTemplateRenderedEvent::class, FilesSharingLoadAdditionalScriptsListener::class);
+		// for image attachments
+		$context->registerEventListener(NodeCopiedEvent::class, NodeCopiedListener::class);
+		$context->registerEventListener(BeforeNodeRenamedEvent::class, BeforeNodeRenamedListener::class);
+		$context->registerEventListener(BeforeNodeDeletedEvent::class, BeforeNodeDeletedListener::class);
```

Comment:
It's not possible to get the attachment dir after a file has been deleted or moved because we use it (the file) to get its owner's storage.
I agree it would be nicer to use NodeRenamedEvent and BeforeNodeDeletedEvent but I think it's not possible.